crates/rattler_solve/src/libsolv/wrapper/pool.rs

```diff
@@ -108,164 +129,96 @@ impl Pool {
             Verbosity::Extreme => 3,
         };
         unsafe {
-            ffi::pool_setdebuglevel(self.0.as_ptr(), verbosity);
+            ffi::pool_setdebuglevel(self.raw_ptr(), verbosity);
         }
     }
 
     /// Set the provided repo to be considered as a source of installed packages
     pub fn set_installed(&self, repo: &Repo) {
```

Comment:
This function can result in undefined behavior if the repo has not been created by the same `Pool`. Maybe you can add an `assert` in here to ensure that Repo belongs to the current Pool?

Reply:
Will do